### ISSUE TITLE
Unable to match multiple values in request body using XPath in SOAP plugin

### ISSUE THREAD

**Issue by ismailastighfar** [outofcoffee/imposter#576]
Description:
I encountered an issue while trying to match multiple values in the request body using XPath in the SOAP plugin. Here are the details:

Problem:
When configuring the Imposter with a SOAP plugin and defining the request body XPath to match multiple values, the matching does not work as expected, i didn't find any examples so if you have any resources or modifications i should do to make this work 

my config file : 

```yaml

resources:  
  - operation: RetrievePatientInfoWithVIPIndicatorRequest
    requestBody:
      xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sInstitutionCode"
      value: "SGH"
      xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sUserID"
      value: "GSRVWCMSTG"
      xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sPatNRIC"
      value: "S2192071A"
      xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sCallerSysADID:"
      value: "WCMS"
      xmlNamespaces:
        env: "http://schemas.xmlsoap.org/soap/envelope/"
        urn: "urn:sap-com:document:sap:rfc:functions"
    response:
      staticFile: ./oas/S2192071A.xml

**Comment by outofcoffee** [comment]
Hi @ismailastighfar, thanks for raising this.

A good fit here is the `allOf` (or `anyOf`) block. For example:

```yaml
resources:  
  - operation: RetrievePatientInfoWithVIPIndicatorRequest
    requestBody:
      allOf:
        - xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sInstitutionCode"
          value: "SGH"
        - xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sUserID"
          value: "GSRVWCMSTG"
        - xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sPatNRIC"
          value: "S2192071A"
        - xPath: "/env:Envelope/env:Body/urn:RetrievePatientInfoWithVIPIndicatorRequest/sCallerSysADID:"
          value: "WCMS"
      xmlNamespaces:
        env: "http://schemas.xmlsoap.org/soap/envelope/"
        urn: "urn:sap-com:document:sap:rfc:functions"
    response:
      staticFile: ./oas/S2192071A.xml
```

More info here: https://docs.imposter.sh/request_matching/#using-multiple-request-body-matchers

**Comment by outofcoffee** [comment]
Hi @ismailastighfar, in the SOAP request, this namespace is declared:

    xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"

Could you try updating the namespace prefix from `soapenv` to `env`? If this works, it may still be a bug we need to fix, but it would be good to have confirmation.

**Comment by ismailastighfar** [comment]
I attempted to follow your suggestion to update the namespace prefix from soapenv to env in the SOAP request. However, I am still experiencing the same issue. 

Response Received
Here's the response I received after updating the namespace prefix as suggested this is the default response if nothing is matching , the same as the namespace soapenv : 

```yaml 
<?xml version="1.0" encoding="UTF-8"?>
<env:Envelope xmlns:env="http://schemas.xmlsoap.org/soap/envelope/">
    <env:Header/>
    <env:Body>
        <urn:PatientNRIC xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:PatientNRIC>
        <urn:PatientNRICType xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:PatientNRICType>
        <urn:PatientName xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:PatientName>
        <urn:PatientBirthDate xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:PatientBirthDate>
        <urn:VIPVVIPIndicator xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:VIPVVIPIndicator>
        <urn:InstitutionSpecificAddress xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:InstitutionSpecificAddress>
        <urn:StatusCode xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:StatusCode>
        <urn:StatusDescription xmlns:urn="urn:sap-com:document:sap:rfc:functions">string</urn:StatusDescription>
    </env:Body>
</env:Envelope>
```

**Comment by outofcoffee** [comment]
Hi @ismailastighfar, thank you for confirming this.

There was a condition that the WSDL parser didn't handle properly - the document wrapped style. This is used in the WSDL you provided, and was resulting in the matching not working.

This should be released in [v3.44.0](https://github.com/outofcoffee/imposter/releases).

Please let us know if this helps.
